webapp/lib/controller.dart

```diff
@@ -150,167 +245,46 @@ Map<String, Map<String, List<String>>> tagData = {
+void fetchConfigurationData() {
+  configurationTagData = configurationData;
+  additionalConfigurationTags = new Set.from(configurationTags);
+  configurationResponseLanguages = new Set.from(configurationReponseLanguageData);
 }
 
-void init() async {
-  view.init();
-  await platform.init();
-}
-
-void initUI() {
-  view.contentView.dashboardView.activePackages.addAll(
-    [
-      new view.ActivePackagesViewPartial('Urgent conversations'),
-      new view.ActivePackagesViewPartial('Open conversations'),
-      new view.ActivePackagesViewPartial('Batch replies (Week 12)'),
-    ]);
-  view.contentView.dashboardView.availablepackages.addAll(
-    [
-      new view.AvailablePackagesViewPartial('Quick Poll',
-        'Ask a question with fixed answers',
-        ['Needs: Q/A, Labelling team, Safeguarding response', 'Produces: Dashboard for distribution of answers']),
-      new view.AvailablePackagesViewPartial('Information Service',
-        'Answer people\'s questions',
-        ['Needs: Response protocol, Labelling team, Safeguarding response', 'Produces: Thematic distribution, work rate tracker']),
-      new view.AvailablePackagesViewPartial('Bulk Message',
-        'Send set of people a once off message',
-        ['Needs: Definition of who. Safeguarding response', 'Produces: Success/Fail tracker'])
-    ]);
-  view.contentView.dashboardView.renderActivePackages();
-  view.contentView.dashboardView.renderAvailablePackages();
-  view.contentView.renderView(view.contentView.dashboardView.dashboardViewElement);
-}
-
-void command(UIAction action, Data actionData) {
-  log.verbose('command => $action : $actionData');
-  switch (action) {
-
-    case UIAction.userSignedIn:
-      initUI();
-      break;
-    case UIAction.goToConfigurator:
-      showConfigurationView();
-      break;
-    case UIAction.configurationTagSelected:
-      ConfigurationTagData data = actionData;
-      retrieveConfigurationTagResponse(data.selectedTag);
-      break;
-    case UIAction.addConfigurationTag:
-      ConfigurationTagData data = actionData;
-      addNewConfigurationTag(data.tagToAdd);
-      break;
-    case UIAction.editConfigurationTagResponse:
-      ConfigurationResponseData data = actionData;
-      updateEditedConfigurationTagResponse(data.parentTag, data.index, data.language, data.text);
-      break;
-    case UIAction.addConfigurationResponseEntries:
-      ConfigurationResponseData data = actionData;
-      addConfigurationResponseEntries(data.parentTag);
-      break;
-  }
-}
-
-void showConfigurationView() {
-  view.contentView.configurationView.tagList.renderTagList(tagData.keys.toList());
-  view.contentView.configurationView.tagResponses.renderResponses(tagData.keys.toList().first, tagData.values.toList().first);
+void showConfigurationView(String selectedTag, Map<String, Map<String, List<String>>> tagData) {
+  if (selectedTag == null) selectedTag = tagData.keys.toList().first;
```

Comment:
in general in Dart, passing thing around that are `null` is a bit frowned upon. Null is often an indication that there is a problem (E.g that something hasn't been initialised properly).

I see that the only place where selectedTag is passed in as null is when handling the `UIAction.goToConfigurator`. I would recommend that you compute the selected tag there the same way (ie. as the first one in the list), and then pass it to this method, instead of having this if here.

Reply:
Understood. Computed the `selectedTag` beforehand